### ISSUE TITLE
should @static work with elseif

### ISSUE THREAD

**Issue by SimonDanisch** [JuliaLang/julia#18861]
Right now, `@static` only evaluates the condition of the first if statement, so any `elseif` won't error, but isn't actually evaluated statically.
Should we clarify the docs, or actually implement `@static` for arbitrary nested conditional statements?
The name does suggests that it just works like a catchall, similar to `@compat`.

**Comment by ararslan** [comment]
Personally I would be in favor of implementing `@static` for arbitrarily nested conditionals. That's actually what I had thought it was doing, and I would guess that others assume that as well.

**Comment by StefanKarpinski** [comment]
I'm in favor of letting the staticness apply to all of an if-elseif-...-else-end structures conditions. That seems like the most intuitive thing that people would expect.

**Comment by yuyichao** [comment]
Unless I'm missing something, it seems a little hard to distinguish that and

```jl
if a
    b
else
    c ? d : e
end
```

**Comment by StefanKarpinski** [comment]
Ah, true. Not sure what to do about it then.

**Comment by TotalVerb** [comment]
Maybe `if x; A; elseif y; B; else; C; end` ought to be parsed as `(if x A (else-if y B C))` instead of `(if x A (if y B C))`, so that these cases can be distinguished.

**Comment by ararslan** [comment]
@yuyichao Would it be bad to statically evaluate that? Seems to me that the ideal behavior of `@static` would be to evaluate _anything_ in its scope, even if it's a ternary nested in an `if`.

**Comment by yuyichao** [comment]
> Would it be bad to statically evaluate that

Yes. Evaluating all nested level will mean that you cannot write any conditional branches in it.

**Comment by ararslan** [comment]
Oh, I see now. Thanks for the explanation.

**Comment by yuyichao** [comment]
If we don't want to break any AST consumer, we can possibly use a meta node to mark the `elseif` in the surface AST.

**Comment by Keno** [comment]
Seems fixed on master:
```
julia> macroexpand(Main, :(@static if false; a; elseif true; b; else; c; end))
quote
    #= REPL[1]:1 =#
    b
end
```
If that was not what the issue was about, needs a better description.
